concepts/code-organization/introduction.md

```diff
@@ -0,0 +1,163 @@
+# Introduction
+
+Big projects often span over several files to keep the code organized in logical components.
+
+## Forward Declarations
+C++ code is evaluated procedurally.
+If you want to use a function, it has to be known to the compiler at the moment of usage.
+Sometimes it is not possible to find a linear fashion to lay out the definitions in the source code.
+Take a look at the example below:
+
+```cpp
+int myFunction(int n) {
+    if (n < 10) {
+        return n;
+    } else {
+        return myOtherFunction(n / 10);
+    }
+}
+
+int myOtherFunction(int m) {
+    return myFunction(m / 2);
+}
+```
+
+When `myFunction` is defined, the compiler does not know about `myOtherFunction` yet.
+Unfortunately, the circle-reference problem cannot be solved by switching the order.
+
+C++ offers **forward declarations** to let the compiler know of `myFunction` and `myOtherFunction` before they are defined.
+The compiler assumes that the definition will follow at some later point after the declaration.
+The next example shows how a forward declaration is used for functions.
+
+```cpp
+int myFunction(int n);          // Forward declaration of myFunction
+int myOtherFunction(int m);     // Forward declaration of myOtherFunction
+
+// Definition of myFunction
+int myFunction(int n) {
+    if (n < 0) {
+        return 0;
+    } else {
+        return myOtherFunction(n - 2);
+    }
+}
+
+// Definition of myOtherFunction
+int myOtherFunction(int m) {
+    return myFunction(m / 2);
+}
+```
+
+Attention: the `;` is needed after the declaration, but not after the definition.
+
+## Header Files
+
+Often, declarations are bundled together in header files. 
+The most common file extension for header files is `.h`.
+Some projects use `.hpp` or skip the extension completely.
+The definitions are located in a separate `.cpp` file.
+
+
+## Includes
+
+It is possible to include the content of other files with the `include` directive.
+Includes should be stated at the top of the file.
+The snippet below shows how to use the `example()` function that was declared in the `myProject.h` file.
+
+```cpp
+#include "myProject.h"
+example();
+```
+
+If the `myProject.h` is located in the `some_folder` and not in the root directory, you would have to use `#include "some_folder/myProject.h"`.
+
+## Include Guards
```

Comment:
> This is a pretty advanced topic since it involves explaining the workings of the preprocessor, in a way. Did you plan to have a preprocessor topic at some point? If so I'd simplify this entry to avoid talking about the specifics and defer to that later topic.

Agreed!